feat/nice_date utils

JarbasAl opened the pull request:
https://github.com/MycroftAI/lingua-franca/issues/198

NeonDaniel commented on lingua_franca/format.py:
```diff
@@ -370,6 +374,39 @@ def nice_date_time(dt, lang='', now=None, use_24hour=False,
 
 
+@localized_function(run_own_code_on=[FunctionNotLocalizedError])
+def nice_day(dt, date_format='MDY', include_month=True, lang=""):
```
perhaps `date_format` should be an enum since there are only 3(?) valid options right now (MDY, DMY, YMD).

JarbasAl replied:
can't hurt, but note this comes from mycroft.conf and the valid options are basically defined there, so let's define what the enum should contain clearly

NeonDaniel reviewed (changes requested):
Added functions need docstrings. If we're not using a standard date formatting string, valid options should either be tied to an enum class or documented in docstrings; unsupported formats should probably raise an exception

emphasize commented on lingua_franca/format.py:
```diff
@@ -370,6 +374,39 @@ def nice_date_time(dt, lang='', now=None, use_24hour=False,
+            return "{} {}".format(month, dt.strftime("%d"))
+        else:
+            return "{} {}".format(dt.strftime("%d"), month)
+    return dt.strftime("%d")
```
just realized that a one digit day is formatted with "0x". I don't know if all TTS can handle that.
Better `dt.strftime("%-d")`

emphasize commented on lingua_franca/format.py:
```diff
@@ -392,6 +429,34 @@ def nice_year(dt, lang='', bc=False):
+    else:
+        raise ValueError("invalid date_format")
+    return {
+        "date_string": dtstr,
```
I feel we should include abbreviations coming from
`nice_month`/`nice_weekday`with a changed signature
```python
def nice_month(dt, abbreviation=False, lang=""):
```
in both cases date_format is irrelevant

emphasize commented on lingua_franca/format.py:
```diff
@@ -370,6 +374,39 @@ def nice_date_time(dt, lang='', now=None, use_24hour=False,
+def nice_weekday(dt, lang=""):
+    if lang in date_time_format.lang_config.keys():
+        localized_day_names = list(
+            date_time_format.lang_config[lang]['weekday'].values())
```
There has to be a `cache` call upon `date_time_format` to be safe to say that the keys are there.

I'm inclined to rewrite the class, that poor class is crammed
object dict created on init, with a helper `get_date_time_format(lang)`

JarbasAl commented:
I'm merging this one and addressing feedback later as this is a direct port of utils used in the datetime skill, 

I want to merge the homescreen PR needing this to reduce the insane amount of log spam it produces, right now im focused on getting 0.0.7 core release out and dont have time to focus on LF
